### DISCUSSION TITLE
How to inject parent state to Scope()?

### DISCUSSION THREAD

**Discussion by Kondamon** [pointfreeco/swift-composable-architecture#2013] [Q&A] [answered]
Hi everyone!

I want to use a variable from the ParentReducer within `Scope`.

The only solution that I found is to create a custom `ChildScope` reducer. In this reducer I could access MyReducer.State and update the dependency of the ChildReducer.

Does anyone know a simpler way, where I don't have to recreate and adapt `Scope` for being able to access state? I don't see a possibility to update reducer dependencies within presentation, e.g. `.fullScreenCover(store: self.store.scope(state: \.$destination,..)`

```Swift
public var body: some ReducerProtocol<State, Action> {
  Reduce<State, Action>(core)
    .ifLet(\MyReducer.State.$destination, action: /MyReducer.Action.destination) {
        Scope(state: /MyReducer.DestinationState.child,
              action: /MyReducer.DestinationAction.child) {
            ChildReducer()
                .dependency(\.provider, .dynamic(state: 
                    // How to get state from MyReducer for dynamic dependency?
                                                                ))
            
        }
      }
}
```

**Comment by lukeredpath** [comment, marked as answer]
You’ll want to use this https://github.com/pointfreeco/swift-composable-architecture/pull/1969
